ante-ls/src/main.rs

```diff
@@ -88,58 +91,265 @@ impl LanguageServer for Backend {
             self.update_diagnostics(params.text_document.uri, &rope).await;
         };
     }
+
+    async fn hover(&self, params: HoverParams) -> Result<Option<Hover>> {
+        self.client.log_message(MessageType::LOG, format!("ante-ls hover: {:?}", params)).await;
+        let uri = params.text_document_position_params.text_document.uri;
+        let rope = match self.document_map.get(&uri) {
+            Some(rope) => rope,
+            None => return Ok(None),
+        };
+
+        let cache = self.create_cache(&uri, &rope);
+        let ast = cache.parse_trees.get_mut(0).unwrap();
```

Comment:
Do we need a map between file names and which parse tree to grab?

Reply:
We're not preserving the cache between each hover, and it get recreated with the file containing the hover location being entry point. As such the 0th parse tree should always contain the file we are interested in.